File: app/src/main/java/com/timotiusoktorio/pencake/di/AppComponent.java
package com.timotiusoktorio.pencake.di;

import android.app.Application;

import com.timotiusoktorio.pencake.data.source.DataManager;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {

    Application getApplication();

    DataManager getDataManager();
}

File: app/src/main/java/com/timotiusoktorio/pencake/di/AppComponent.kt
package com.timotiusoktorio.pencake.di

import android.app.Application
import com.timotiusoktorio.pencake.data.source.DataManager
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [(AppModule::class)])
interface AppComponent {

    val application: Application

    val dataManager: DataManager
}